enterprise/ha/src/main/java/org/neo4j/kernel/ha/cluster/HighAvailabilityModeSwitcher.java

```diff
@@ -311,8 +311,10 @@ public void run()
                     haCommunicationLife.shutdown();
                     haCommunicationLife = new LifeSupport();
 
+                    // it is important for availableMasterId to be re-read on every attempt so that
+                    // slave switching would not result in an infinite loop with wrong/stale availableMasterId
                     URI resultingSlaveHaURI =
-                            switchToSlave.switchToSlave( haCommunicationLife, me, masterUri, cancellationHandle );
+                            switchToSlave.switchToSlave( haCommunicationLife, me, availableMasterId, cancellationHandle );
```

Comment:
Nice fix. A minor detail: If you want you should probably remove `masterUri` method-local variable altogether so that there will not be a mistake in code-completion or whatever in the future.